src/util/versionHandlers.ts

```diff
@@ -0,0 +1,131 @@
+/* 
+Performs actions dependent on the trajectory file format version.
+Currently supported versions: 1, 2
+*/
+
+import {
+    TrajectoryFileInfo,
+    TrajectoryFileInfoV1,
+    TrajectoryFileInfoV2,
+} from "@aics/simularium-viewer/type-declarations/simularium";
+const si = require("si-prefix");
+
+import { TimeUnits } from "../state/metadata/types";
+
+const invalidVersionNumberError =
+    "Invalid version number in TrajectoryFileInfo";
+
```

Comment:
This is definitely clearer this way, and will be easier to update. I have one more organization idea I want to throw out there, Dan might have additional thoughts too, but after seeing this file and all the switch statements it seems like it would be nice to have all the functions organized by version number, something like this (this could also be a class, but I thought just doing it in objects would be clear): 
```JavaScript
// an object that holds all the functions organized by version number, functions would mostly have the same names, just different logic
const versionMapping = {
    1: {
        function1: (data) => { return processedData }, // pseudo code, just "do something to the data"
        function2: (data) => { return processedData },
    },
    2: {
        function1: (data) => { return processedData },
        function2: (data) => { return processedData },
        function3: (data) => { return processedData },
    },
};

const versionManager = {
    set version(version) { this.version = version},
    functionCheck(functionName) {
       if (!this.version) {
            throw "Need a version number";
        }
        else if (!versionMapping[this.version][functionName]) {
            throw "Function doesn't exist for version"
        }
    }

    get function1() {
        functionCheck(function1);
        versionMapping[this.version].function1;
    }
    get function2() {
        functionCheck(function1);
        versionMapping[this.version].function2;
    }
    get function3() {
        functionCheck(function1);
        versionMapping[this.version].function2;
    }
}
// how you would use it to call a function (`data` is the unprocessed data coming from the viewer):
versionManager.version = data.version;
const processedData = versionManager.function1(data);```

Reply:
Interesting suggestion! I'd like to hear what @toloudis has to say, too. I wonder if the best solution _would_ be to convert all old data to the latest data before doing anything with it like he mentioned before. And if so where that should happen. If we're going to drop support for version 1 soon, all of this seems overkill, but I know we need to think about this for future versions too.

Reply:
I have to catch up and reread this, but my basic thought is:  at the point when data FIRST arrives (in the viewer code), figure out what version it is.  If it's not the CURRENT version, then convert it on the spot and pass it on as current version data.  Then everything downstream only deals with current version, and the version check only exists in one single place, at the very first point where the data is read.  ( If doing that causes excessive expensive copying, then it might be a performance/memory hit but I doubt that it is, for this data here. )

Reply:
To illustrate, this JS library does something like what I'm describing:  https://github.com/TomFrost/Vers
I'm not saying we should use that library (maybe it's good, idk) but it shows the type of pattern I am thinking about